src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/BackgroundAspectEffect/BackgroundAspectEffectRouter.ios.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using UIKit;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+using Effects = Xamarin.CommunityToolkit.iOS.Effects;
+
+[assembly: ExportEffect(typeof(Effects.BackgroundAspectEffectRouter), nameof(BackgroundAspectEffectRouter))]
+
+namespace Xamarin.CommunityToolkit.iOS.Effects
+{
+	public class BackgroundAspectEffectRouter : PlatformEffect
+	{
+		UIImageView? imageView;
+
+		CancellationTokenSource cancellationTokenSource;
+
+		public BackgroundAspectEffectRouter()
+		{
+			cancellationTokenSource = new CancellationTokenSource();
+		}
+
+		protected override void OnAttached()
+			=> Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource.Token);
+
+		protected override void OnDetached()
+			=> ClearBackgroundImage();
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (!args.PropertyName.Equals(BackgroundAspectEffect.AspectProperty.PropertyName))
+				return;
+
+			cancellationTokenSource.Cancel();
+
+			Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource.Token);
+		}
+
+		async Task ApplyBackgroundImageAsync()
+		{
+			var contentPage = Element as ContentPage;
+
+			if (Element is null || contentPage is null || contentPage.BackgroundImageSource is null)
+				return;
+
+			var image = await GetBackgroundImageFromSource(contentPage.BackgroundImageSource);
+
+			if (image is null)
+				return;
+
+			var fallbackColor = contentPage.BackgroundColor.ToUIColor();
+
+			Device.BeginInvokeOnMainThread(() => ApplyImageToView(image, fallbackColor));
+		}
+
+		void ApplyImageToView(UIImage image, UIColor color)
+		{
+			var effectParam = BackgroundAspectEffect.GetAspect(Element);
+
+			imageView = CreateImageView(image, effectParam);
+
+			Container.BackgroundColor = color;
+
+			Container.InsertSubview(imageView, 0);
+
+			ApplyConstraints(imageView, Container);
+		}
+
+		void ClearBackgroundImage()
+		{
+			if (imageView != null)
+			{
+				imageView.RemoveFromSuperview();
+
+				imageView.Dispose();
+			}
+
+			// TODO: Is this enough to cleanup?
+			cancellationTokenSource.Cancel();
+			cancellationTokenSource.Dispose();
```

Comment:
Setting it to null, just to be safe.
```suggestion
			cancellationTokenSource.Dispose();
			cancellationTokenSource = null;
```